lib/galaxy/datatypes/interval.py

```diff
@@ -1482,6 +1482,86 @@ def sniff( self, filename ):
         return False
 
 
+class ScIdx(Tabular):
+    """
+    Support for the Strand-specific coordinate count datatype which is used
+    by tools within the Chip-exo Galaxy flavor.  The Chip-exo Galaxy flavor
+    is used by the Center for Eukaryotic Gene Regulation labs at Penn State
+    University.
+
+    ScIdx files are 1-based and consist of strand-specific coordinate counts.
+    They always have 5 columns, and the first row is the column labels:
+    'chrom', 'index', 'forward', 'reverse', 'value'.
+    Each line following the first consists of data:
+    chromosome name (type str), peak index (type int), Forward strand peak
+    count (type int), Reverse strand peak count (type int) and value (type int).
+    The value of the 5th 'value' column is the sum of the forward and end
+    reverse peak count values.
+    """
+    file_ext = "scidx"
+
+    MetadataElement(name="columns", default=0, desc="Number of columns", readonly=True, visible=False)
+    MetadataElement(name="column_types", default=[], param=metadata.ColumnTypesParameter, desc="Column types", readonly=True, visible=False, no_value=[])
+
+    def __init__(self, **kwd):
+        """
+        Initialize scidx datatype.
+        """
+        Tabular.__init__(self, **kwd)
+        # Don't set column names since the first
+        # line of the dataset displays them.
+        self.column_names = []
+
+    def sniff(self, filename):
+        """
+        Checks for 'scidx-ness.'
+        """
+
+        def is_scidx(fh, bool):
```

Comment:
@gregvonkuster can you use try, (except), finally instead? https://docs.python.org/2.5/whatsnew/pep-341.html